### ISSUE TITLE
Add `noopener-allow-popups` value to Opener Policy

### ISSUE THREAD

**Issue by AtkinsSJ** [LadybirdBrowser/ladybird#1690]
Spec change is here: https://github.com/whatwg/html/pull/10394, which updates [this page](https://html.spec.whatwg.org/multipage/browsers.html) of the HTML spec.

**Comment by SuhasHareesh** [comment]
Hi, this is my first foray into open source contribution. Is this issue still open? If so, can I work on it?

**Comment by Psychpsyo** [comment]
> Hi, this is my first foray into open source contribution. Is this issue still open? If so, can I work on it?

It is still open (literally), even though part of it has been implemented.
Note that the person who did that also said [they are working on the rest of that implementation](https://github.com/LadybirdBrowser/ladybird/pull/3663#issuecomment-2687786210), so it might not be the best issue to start working on, given that someone already seems to be doing it.

That's generally the main thing to look out for when choosing which issue to tackle.
(other things to pay attention to might be how complex the issue is and if there's any debate over how/if it should actually be implemented, but that is quite rare)

**Comment by SuhasHareesh** [comment]
@Psychpsyo Thank you for your real eye-opening advice!! I'll keep this in mind! :)

**Comment by nhlmg93** [comment]
@AtkinsSJ  is there a commit or pr that would be a good example of whats is trying to be achieved here?

**Comment by AtkinsSJ** [comment]
> is there a commit or pr that would be a good example of whats is trying to be achieved here?

Not really. I wrote a little explanation here: https://github.com/LadybirdBrowser/ladybird/pull/3663#issuecomment-2687524925

Basically: look at the spec PR, see what changed, and make those changes in our code.

**Comment by nhlmg93** [comment]
It would appear this does not even work in the chrome browser.

i modeled my test from the scenario described in the link below 
 https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Cross-Origin-Opener-Policy#severing_the_opener_relationship 

index.ts
`
import { Hono } from "hono";
import { serveStatic } from "hono/bun";
import { cors } from "hono/cors";

const app = new Hono();

app.use(
  "*",
  cors({
    origin: "*", // Allow all origins. In production, you might want to restrict this
    allowMethods: ["GET", "POST", "PUT", "DELETE", "OPTIONS"],
    allowHeaders: ["Content-Type", "Authorization"],
    exposeHeaders: ["Content-Length"],
    maxAge: 86400,
    credentials: true,
  }),
);

app.get("/", serveStatic({ path: "./index.html" }));
app.get(
  "/test/",
  (c, next) => {
    c.header("Content-Type", "text/html");
    c.header("Cross-Origin-Opener-Policy", "noopener-allow-popup");
    return next();
  },
  serveStatic({ path: "./public/test.html" }),
);

index.html
`
<h1>Index</h1>

<script>
  // Open a new window
  const handle = window.open("http://localhost:3000/test/", "secureTab");
  if (handle.closed) {
    console.log("window closed");
  }
</script>
`

test.html
`
<h1>Hello, world!</h1>
`

// Start the server
const port = process.env.PORT || 3000;
console.log(`Server is running on port ${port}`);

export default {
  port,
  fetch: app.fetch,
};
`
OUTCOME OF TEST IN CHROME BROWSER:

when navigating to the "/" route i get a popup however i am still able to close(via scripting) the new popup/window from the "opener window". The opener should not be able to close it. and the opener attribute closed should be true. I am trying to figure out expected behavior and set up a test scenario in which i can play around with. Can you see anything wrong above that would make the proxy stay around?

**Comment by nhlmg93** [comment]
> > is there a commit or pr that would be a good example of whats is trying to be achieved here?
> 
> Not really. I wrote a little explanation here: [#3663 (comment)](https://github.com/LadybirdBrowser/ladybird/pull/3663#issuecomment-2687524925)
> 
> Basically: look at the spec PR, see what changed, and make those changes in our code.

greping for #matching-coop in ladybird does not return anything.

```
grep -r "#matching-coop" .
```
command i ran while in ladybird root. returned nothing.

""
grep -r "#cross-origin-opener-policy"
```

returns as i expected 
```
./Libraries/LibWeb/HTML/CrossOrigin/OpenerPolicy.h:// https://html.spec.whatwg.org/multipage/origin.html#cross-origin-opener-policy-value
./Libraries/LibWeb/HTML/CrossOrigin/OpenerPolicy.h:// https://html.spec.whatwg.org/multipage/origin.html#cross-origin-opener-policy
```

**Comment by AtkinsSJ** [comment]
I made some wrong assumptions while making this issue. 😅 

I just tried doing this myself, as it's clearly not appropriate as a good first issue. And it turns out that the changes are all in algorithms that we don't implement. So there's nothing specifically to do here. Oops.

Pinging @alicelee2735 in case you were also looking into this.

**Comment by alicelee2735** [comment]
> I made some wrong assumptions while making this issue. 😅
> 
> I just tried doing this myself, as it's clearly not appropriate as a good first issue. And it turns out that the changes are all in algorithms that we don't implement. So there's nothing specifically to do here. Oops.
> 
> Pinging [@alicelee2735](https://github.com/alicelee2735) in case you were also looking into this.

@AtkinsSJ Thanks for the ping and the update! Since there's nothing to do now, I'll keep an eye out for other good first issues.
